File: src/Microsoft.Management.Deployment.Client/PackageVersionInfo.cpp
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
#include "pch.h"
#include <PackageVersionInfo.h>
#include "PackageVersionInfo.g.cpp"

namespace winrt::Microsoft::Management::Deployment::implementation
{
    hstring PackageVersionInfo::GetMetadata(winrt::Microsoft::Management::Deployment::PackageVersionMetadataField const&)
    {
        throw hresult_not_implemented();
    }
    hstring PackageVersionInfo::Id()
    {
        throw hresult_not_implemented();
    }
    hstring PackageVersionInfo::DisplayName()
    {
        throw hresult_not_implemented();
    }
    hstring PackageVersionInfo::Version()
    {
        throw hresult_not_implemented();
    }
    hstring PackageVersionInfo::Channel()
    {
        throw hresult_not_implemented();
    }
    winrt::Windows::Foundation::Collections::IVectorView<hstring> PackageVersionInfo::PackageFamilyNames()
    {
        throw hresult_not_implemented();
    }
    winrt::Windows::Foundation::Collections::IVectorView<hstring> PackageVersionInfo::ProductCodes()
    {
        throw hresult_not_implemented();
    }
    winrt::Microsoft::Management::Deployment::PackageCatalog PackageVersionInfo::PackageCatalog()
    {
        throw hresult_not_implemented();
    }
    winrt::Microsoft::Management::Deployment::CompareResult PackageVersionInfo::CompareToVersion(hstring versionString)
    {
        throw hresult_not_implemented();
    }
}

File: src/PowerShell/Microsoft.WinGet.Client/PSObjects/PackageVersionInfo.cs
﻿// -----------------------------------------------------------------------------
// <copyright file="PackageVersionInfo.cs" company="Microsoft Corporation">
//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
// </copyright>
// -----------------------------------------------------------------------------

namespace Microsoft.WinGet.Client.PSObjects
{
    using System.Linq;

    /// <summary>
    /// CatalogPackage wrapper object for displaying to PowerShell.
    /// </summary>
    public class PackageVersionInfo
    {
        private Management.Deployment.PackageVersionInfo packageVersionInfo;

        /// <summary>
        /// Initializes a new instance of the <see cref="PackageVersionInfo"/> class.
        /// </summary>
        /// <param name="packageVersionInfo">PackageVersionInfo COM object.</param>
        public PackageVersionInfo(Management.Deployment.PackageVersionInfo packageVersionInfo)
        {
            this.packageVersionInfo = packageVersionInfo;
        }

        /// <summary>
        /// Gets the name of the package version info.
        /// </summary>
        public string DisplayName
        {
            get
            {
                return this.packageVersionInfo.DisplayName;
            }
        }

        /// <summary>
        /// Gets the id of the package version info.
        /// </summary>
        public string Id
        {
            get
            {
                return this.packageVersionInfo.Id;
            }
        }

        /// <summary>
        /// Gets the publisher of the package version info.
        /// </summary>
        public string Publisher
        {
            get
            {
                return this.packageVersionInfo.Publisher;
            }
        }

        /// <summary>
        /// Gets the channel of the package version info.
        /// </summary>
        public string Channel
        {
            get
            {
                return this.packageVersionInfo.Channel;
            }
        }

        /// <summary>
        /// Gets the list of package family names of the package version info.
        /// </summary>
        public string[] PackageFamilyNames
        {
            get
            {
                return this.packageVersionInfo.PackageFamilyNames.ToArray();
            }
        }

        /// <summary>
        /// Gets the list of product codes of the package version info.
        /// </summary>
        public string[] ProductCodes
        {
            get
            {
                return this.packageVersionInfo.ProductCodes.ToArray();
            }
        }

        /// <summary>
        /// Compares the version string with the package version info and returns the CompareResult.
        /// </summary>
        /// <param name="version">Version string.</param>
        /// <returns>CompareResult string.</returns>
        public string CompareToVersion(string version)
        {
            return this.packageVersionInfo.CompareToVersion(version).ToString();
        }
    }
}
